modules/workergroup/README.md

```diff
@@ -0,0 +1,141 @@
+# Worker groups
+
+This sub-module supports different modes of OKE worker node management with advanced configuration.
+
+## Usage
+Worker groups are configured with the optional `worker_groups` array.
+
+The module is implemented in the parent module for general use, and can also be used directly by running `terraform init/apply` in this directory or with a separate state. In this case, the `workergroup` module adds pools to an existing cluster.
+
+1. `cp vars-profile.auto.tfvars.example vars-profile.auto.tfvars`
+1. Define `Required parameters`
+1. Review defaults and `worker_groups`
+1. `terraform init`, `terraform apply` - *update *
+1. *e.g.* change `worker_groups[x]` *size from `1` -> `10` - scale up group*
+1. *e.g.* change `worker_groups[y]` *size from `1` -> `0` - suspend group, retain instance configuration, pool resources*
+1. *e.g.* change `worker_groups[z]` *create from `true` -> `false` - destroy group, retain definition*
+1. *e.g.* add `worker_groups[a]` - *create new group*
+1. *e.g.* remove `worker_groups[x]` - *destroy group*
+1. *e.g.* `terraform refresh` *update convenience variables e.g. worker_primary_ips*
+
+### Mode
+The `mode` parameter determines the mechanism used to provision and manage nodes in the worker group. Currently the only available mode is `node-pool`.
+
+#### **`node-pool`** _(default)_
+See [Scaling Kubernetes Clusters and Node Pools](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengscalingkubernetesclustersnodepools.htm) for more information.
+```yaml
+{
+    name             = "np-vm",
+    mode             = "node-pool",
+    image_id         = "ocid1.image.oc1.phx.aaaaaaaa",
+    shape            = "VM.Standard.E4.Flex",
+    ocpus            = 2,
+    memory           = 16,
+    boot_volume_size = 150,
+    size             = 1,
+}
+```
+
+### Defaults
+Many parameters to a worker group can be defined at multiple levels, taken in priority: `Group > Variable > built-in default`. This enables sparse definition of worker groups that share many traits.
+```yaml
+label_prefix                   = "dev"
+worker_group_create            = true
+worker_group_size              = 0
+worker_group_image_id          = "ocid1.image..." # Required here and/or on group
+worker_group_mode              = "node-pool"
+worker_group_shape             = "VM.Standard.E4.Flex"
+worker_group_ocpus             = 2
+worker_group_memory            = 16
+worker_group_boot_volume_size  = 100
+worker_nsg_ids                 = []
+worker_compartment_id          = "" # Defaults to compartment_id when empty
+worker_group_primary_subnet_id = "" # Defaults to Terraform-managed when empty
+worker_groups                  = [
```

Comment:
It looks like the worker groups are defined in a list. This is different from node pools which is a map. 

Questions:
1. Is there any reason why you selected this?
2. Is it possible to keep map as the same data structure?

The reason we used map for node pools is to facilitate with upgrade of node pools. Although the worker nodes don't have identity, a node pool does have identity. This way, during an upgrade, a user can select exactly which pool to upgrade. With a list, this is not possible.